src/MorseL.Sockets/IWebSocketConnectionManager.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Net.WebSockets;
+using System.Threading.Tasks;
+
+namespace MorseL.Sockets
+{
+    public interface IWebSocketConnectionManager
```

Comment:
How much of a lift is it to turn this into IConnectionManager and get rid of the WebSocket dependency?

Reply:
Potentially IConnectionManager<TPipeline> and then WebSocketConnectionManager<Socket>?

Reply:
It would largely depend on if `WebSocket` inherits from a base socket we could use. Otherwise we'll likely have to do a wrapper around it and use some sort of `ISocket` abstraction. That probably isn't too much work since we _only_ support web sockets.

Reply:
I made #52 for this.